shard.go

```diff
@@ -0,0 +1,97 @@
+package wavelet
+
+import (
+	"sync"
+)
+
+// TransactionMap is a sharded map with TransactionID as its key
+type TransactionMap map[uint64]*transactionMapShard
+
+type transactionMapShard struct {
+	items map[TransactionID]interface{}
+	lock  *sync.RWMutex
```

Comment:
Wonder if we should use a pure `sync.Mutex` instead. For large number of readers `sync.RWMutex` would make sense, though the time it takes to lock/unlock for a `sync.RWMutex` in general is about 3 - 4x slower.

Reply:
I don't see a need for a pointer to a mutex, just the mutex being in the struct is enough.